code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -24,64 +27,57 @@
                 .select(ACCOUNT.fields())
-                    .from(ACCOUNT)
+                .from(ACCOUNT)
                 .join(CREDENTIAL)
-                    .on(CREDENTIAL.ACCOUNT_ID.eq(ACCOUNT.ID))
-                .where(
-                    ACCOUNT.STATE.eq(CommonStatesConstants.ACTIVE)
-                    .and(CREDENTIAL.STATE.eq(CommonStatesConstants.ACTIVE))
-                    .and(CREDENTIAL.PUBLIC_VALUE.eq(access))
-                    .and(CREDENTIAL.SECRET_VALUE.eq(secretKey)))
-                    .and(CREDENTIAL.KIND.in(SUPPORTED_TYPES.get()))
+                .on(CREDENTIAL.ACCOUNT_ID.eq(ACCOUNT.ID))
+                .where(ACCOUNT.STATE.eq(CommonStatesConstants.ACTIVE).and(CREDENTIAL.STATE.eq(CommonStatesConstants.ACTIVE))
+                        .and(CREDENTIAL.PUBLIC_VALUE.eq(access)).and(CREDENTIAL.SECRET_VALUE.eq(secretKey))).and(CREDENTIAL.KIND.in(SUPPORTED_TYPES.get()))
                 .fetchOneInto(AccountRecord.class);
     }
 
     @Override
     public Account getAccountByExternalId(String externalId, String externalType) {
-        return create()
-                .selectFrom(ACCOUNT)
-                .where(
-                        ACCOUNT.EXTERNAL_ID.eq(externalId)
-                        .and(ACCOUNT.EXTERNAL_ID_TYPE.eq(externalType))
-                        .and(ACCOUNT.STATE.eq("active"))
-                ).orderBy(ACCOUNT.ID.asc()).limit(1).fetchOne();
+        return create().selectFrom(ACCOUNT)
+                .where(ACCOUNT.EXTERNAL_ID.eq(externalId).and(ACCOUNT.EXTERNAL_ID_TYPE.eq(externalType)).and(ACCOUNT.STATE.eq("active")))
+                .orderBy(ACCOUNT.ID.asc()).limit(1).fetchOne();
     }
-    
+
     @Override
     public Account createAccount(String name, String kind, String externalId, String externalType) {
-        Map<String, Object> properties =  new HashMap<>();
+        Map<String, Object> properties = new HashMap<>();
         properties.put(ACCOUNT.NAME.toString(), name);
         properties.put(ACCOUNT.KIND.toString(), kind);
         properties.put(ACCOUNT.EXTERNAL_ID.toString(), externalId);
         properties.put(ACCOUNT.EXTERNAL_ID_TYPE.toString(), externalType);
         return resourceDao.createAndSchedule(Account.class, properties);
     }
-    
+
     @Override
     public Account getAccountByUuid(String uuid) {
-        return create()
-                .selectFrom(ACCOUNT)
-                .where(
-                        ACCOUNT.UUID.eq(uuid)
-                        .and(ACCOUNT.STATE.eq("active"))
-                        ).orderBy(ACCOUNT.ID.asc()).limit(1).fetchOne();
+        return create().selectFrom(ACCOUNT).where(ACCOUNT.UUID.eq(uuid).and(ACCOUNT.STATE.eq("active"))).orderBy(ACCOUNT.ID.asc()).limit(1).fetchOne();
     }
-    
+
+    @Override
+    public int updateAccountByKind(String name, String kind, String externalId, String externalType) {
+        Map<TableField<AccountRecord, String>, String> properties = new HashMap<>();
```

Comment:
Whoa!!!  This is will update ALL accounts of a certain kind.   Are you sure you want that?  I'm guessing no.  It seem like this method shouldn't exist.